iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -344,7 +344,7 @@ private async Task ConnectAsync(IChannelHandlerContext context)
                 // the SDK will enable plug and play features by appending the model Id to the MQTT CONNECT packet (in the username).
                 if (!(_options?.ModelId).IsNullOrWhiteSpace())
                 {
-                    usernameString += $"&{ModelIdParam}={_options.ModelId}";
+                    usernameString += $"&{ModelIdParam}={Uri.EscapeDataString(_options.ModelId)}";
```

Comment:
should we escape just the querysting?
I would not expect `dtmi:rido:model;1?hash=C32C4` to be encoded as `dtmi%3Arido%3Amodel%3B1%3Fhash%3DC32C4` rather `dtmi:rido:model;1%3Fhash%3DC32C4`

Reply:
What's the value look like for Amqp?

Reply:
As long as it is unencoded on the service side, it shouldn't matter how it is encoded over the wire.

Reply:
But I'm taking the question seriously and making sure we're doing the right thing. My understanding is that the whole value is being escaped in the C SDK.

Reply:
Confirmed whole value is escaped in the C SDK, and arrives at the Hub twin intact (aka unescaped). I'll confirm for this change and then merge.

Reply:
Yep, works fine.